src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,158 @@
 package frc.team2412.robot.subsystem;
 
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.TalonFXFeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
-
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterSubsystem extends SubsystemBase {
-    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor, WPI_TalonFX hoodMotor) {
+    // instance variables
+    private final WPI_TalonFX flywheelMotor1;
+    private final WPI_TalonFX flywheelMotor2;
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    // Constants
+    public static final double FLYWHEEL_VELOCITY = 10;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = -180; // Total ~360 degrees of rotation, assumes 0 is center
+    public static final double MAX_TURRET_ANGLE = 180;
+    public static final double TURRET_MAX_SPEED = 0.1;
+    public static final double TURRET_MIN_SPEED = -0.1;
+    public static final double TURRET_P = 0.01; // Placeholder PID constants
+    public static final double TURRET_I = 0;
+    public static final double TURRET_D = 0;
+    public static final double MAX_HOOD_ANGLE = 40.0;
+    public static final double MIN_HOOD_ANGLE = 5;
+
+    /**
+     * Constructor for shooter subsystem.
+     * 
+     * @param flywheelMotor1
+     *            The first motor connected to the flywheel
+     * 
+     * @param flywheelMotor2
+     *            The second motor connected to the flywheel
+     * 
+     * @param turretMotor
+     *            The motor that controls the horizontal rotation of the
+     *            turret
+     * 
+     * @param hoodMotor
+     *            The motor that controls the angle of the hood above the
+     *            turret
+     * 
+     */
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        // Motor configs
+        var flywheelLimit = new SupplyCurrentLimitConfiguration(true, 40, 40, 500);
+        flywheelMotor1.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor1.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor2.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.setInverted(true);
+
+        SupplyCurrentLimitConfiguration currentLimit = new SupplyCurrentLimitConfiguration(true, 10, 10, 500);
```

Comment:
Update naming to turretLimit to follow precedent above. Also, these are both constants and should be moved to constants area (Consider making a constants subclass like intake)

Reply:
We used "currentLimit" because we used the same limit for both turret and hood, now the names are "flywheelCurrentLimit", "turretCurrentLimit", and "hoodCurrentLimit" (directly assigned to `turretCurrentLimit`).